### PR TITLE
Prep for v4.10 and v5.0.1

### PR DESCRIPTION
**Pull request by yadij**

### CODE REVIEW

**Comment by yadij** [comment]
@eduard-bagdasaryan can you check why Semaphore is complaining about the end of file on v5 release notes? this PR is not touching that area of the file.

**Comment by eduard-bagdasaryan** [comment]
I see that Semaphore is complaining about a whitespace error (when performing git diff --check):

```
doc/release-notes/release-5.sgml:131: trailing whitespace.
```

**Comment by yadij** [comment]
Yeah, but that is the end of the file which is not being touched by this change.

File: doc/release-notes/release-5.sgml [hunk 1]
```diff
@@ -122,6 +124,35 @@ Most user-facing changes are reflected in squid.conf (see below).
    the helpers functionality to rebuild it as needed.
 
 
+<sect1>Loop Detection in Content Delivery Networks
+<p>Details in <url url="https://tools.ietf.org/html/rfc8586" name="RFC 8586">
+
+<p>Squid now uses the CDN-Loop header as a source for loop detection.
+
```
Comment (eduard-bagdasaryan) [inline doc/release-notes/release-5.sgml]: > Yeah, but that is the end of the file which is not being touched by this change.

@yadij , it complains about this line (131), which was modified and is not the end of the line.

**Comment by yadij** [comment]
Doh. I misread the report. Thanks.

**Comment by rousskov** [comment]
> yadij assigned rousskov and kinkie yesterday

@yadij, please do not assign PRs to me. I view PR assignment as a responsibility for the PR as a whole. As you know, I routinely help with PRs, but since you are not my manager, you cannot decide what PRs I should be responsible for.

**Comment by yadij** [comment]
@rousskov who then would you say is responsible for this PR?  I am the author here.

**Comment by rousskov** [comment]
> who then would you say is responsible for this PR?

You are.

> I am the author here.

You seem to imply that an author cannot be responsible for their own PR, but that implication is false. For the vast majority of Squid PRs, the author is (or should be) responsible for the PR overall.

**Comment by yadij** [comment]
I was asking in context of your complaint that I was abusing my maintainer privileges merging PRs without QA approval first. The trigger appears to be a miscommunication about changes to the v5 branch status - but still taken your pain into consideration (and fully agree in regards to master PR commits) and we have a good test case here to sort out the boundaries.

The needs that are being met are the bi-monthly v4 release (4.10) which was due Jan 6th. You indicated earlier about my release schedule "nobody follows that schedule", which was wrong. The downstream distributors and I follow that schedule. *Nobody else* needs to, though there are some admin doing local custom builds who do follow anyway. If I miss a release deadline there is a small flow of private email asking what is going on - the first such mails about 4.10 are arriving already.

Both PR 538 and PR 537 have to be merged for 4.10 (and 5.0.1) to be released.

As author I cannot put an explicit approve on this PR 538 (github restriction). The QA system automatically detects author as one vote positive (so good that far). As squidadm scripts authors the PR 537 I can and have signed off on the changes there.

Who else is there to be responsible for this PR getting merged in less than 10 days** (2 votes required to fast track)? It is unfortunate indeed that we have such a lack of QA personnel, but that is what we have to deal with for now.


** The dates entered into the ChangeLog factored in the Anubis 48hr fast-track delay plus time for the backports testing delays. But still requires either a) someone else in QA to sign off approval, or b) waive maintainer privileges at github. 

I believe we would both rather have the first (a) happen, which is why I pinged *both* you and kinkie for followup reviews. But if this takes many days I will waive privileges at these non-code release PRs. Sadly my next free days to work on Squid start tomorrow so I will effectively be forced to do that waiving in 24hrs if nobody else in QA has time to handle it properly beforehand.

**Comment by rousskov** [comment]
> I was asking in context of your complaint that I was abusing my maintainer privileges merging PRs without QA approval first.

That is not what I was complaining about. If you would like to discuss specifics, let's do it via email or on the IRC.

> You indicated earlier about my release schedule "nobody follows that schedule", which was wrong.
 
AFAICT, that schedule has not been followed but, again, if you would like to discuss specifics, let's do it elsewhere.

> Who else is there to be responsible for this PR getting merged in less than 10 days?

IIRC, anybody from https://github.com/orgs/squid-cache/teams/commiters can add a second vote (currently four people other than you). You (as the PR author) are responsible for making it happen (if that is what you want to happen).

Needless to say, you can certainly ask folks to review PRs. GitHub even has a built-in mechanism for such review requests (but you do not have to use that mechanism -- you can just add a comment asking folks to review).

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,64 @@
+Changes to squid-5.0.1 (08 Jan 2020):
```
Comment (rousskov) [inline ChangeLog]: What is "squid-5.0.1"? I see no corresponding git tag. Judging by the number of entries in this list, the v5.0.1 tag would have to be pointing to some commit made long time ago, which does not make sense. Perhaps this should say 

```suggestion
Changes in squid-5.0.1 (08 Jan 2020):
```

or

```suggestion
Changes up to squid-5.0.1 (08 Jan 2020):
```


instead?

Comment (rousskov) [reply ChangeLog]: I edited the previous comment to make the precondition for my suggestions explicit and to express a preference for the first suggestion.

Comment (yadij) [reply ChangeLog]: The text is what Duane picked for use. It only matter to software parsing our ChangeLog.

In regards to the tags, the backport commit of this PR to the v* branch in the squidadm repository is where the tag is applied and the release bundle is generated from that commit.

**Comment by yadij** [comment]
> Should we adjust the release process so that in-version releases (i.e. anything that happens on a numbered branch like `v4`) do not require master branch modification?
> 

If you really want to go that way I can work on fully splitting the release process from master and having these prep and release commit changes forward-ported by through the maintenance enforcement PRs. That latter part would still be necessary to maintain the sources integrity on master for testers building that code directly. It would somewhat simplify the automation though to have a virtual v* branch that simply bumped forward along master daily like v5 did these past few months for whatever the next series is planned to be.

**Comment by rousskov** [comment]
> Alex: Should we adjust the release process so that in-version releases do not require master branch modification?

> Amos: If you really want to go that way

It is a suggestion aimed at reducing (more expensive, on many levels) master changes or, at the very least, eliminating their urgency. This idea needs _maintainer_ (i.e. your) buy-in:

* If you think it is a good idea that you would be happy to implement, then let's discuss how the master branch will gain versioned ChangeLog entries. That discussion should probably start with what synchronization invariants we want to maintain in master (since we will be violating the current "versioned branches use the corresponding pieces of master's ChangeLog" invariant).
* Otherwise, let's ignore it.

> It would somewhat simplify the automation though to have a virtual v* branch that simply bumped forward along master daily like v5 did these past few months for whatever the next series is planned to be.

This simplification seems to be unrelated to this PR and my suggestion (i.e., copying versioned branch ChangeLog entries into master rather than the other way around). If you want to pursue the virtual branch idea, we should probably discuss it elsewhere, but here are a few quick unordered notes (that might make such a discussion unnecessary):

* Official branches carry a lot of "baggage" for a lot of people. Script simplification is probably the last goal that should be driving their creation.
* If you need a virtual branch to simplify scripts, the scripts should create that branch (in the repository used by the scripts). No need to pollute the official repository to satisfy this internal need!
* If this virtual branch exists, then it may be difficult to justify having the master branch. There are probably projects out there that successfully use version branches instead of master, but I am not sure we want to introduce this surprise into the Squid project (now; without careful analysis that we do not have the resources for).
* If virtual branch always points to the tip of the master, then you need scripts to keep the virtual branch updated (so the net result may end up being more or more complex scripts). If virtual branch does not always point to the tip of the master, then we need to decide what that virtual branch really represents (e.g., The last daily snapshot? That should be a tag, not a branch. Something else?).

**Pull request closed without merging**